strapi/strapi
# Add code generators

### What does it do?

It adds `plop` generators for all plugins extending the admin panel (currently only generate components).

`plop` is installed as a dev dependency of the root workspace, if we want to make it available to plugin developers we'll need to change its installation.

The files are located under `generators/admin` for now, but it should probably be moved elsewhere since we'll make other generators unrelated to the admin panel. I'm listening to suggestion :)

### Why is it needed?

It should make the development of plugins easier.

### How to test it?

Make sure your packages are installed with `yarn` then run `yarn generate component` and respond to the given prompts :)

- `useReducer` template is missing and other generators will come soon (probably in another PR)
